### PR TITLE
Feature/virtualizer new approach

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This PR is still in draft and before merging it we should clean a bit the code and improve the performances on iCub

This PR is only to keep track of the activity.  This code runs in icub-virtualizer laptop

### CODE REVIEW

**Comment by GiulioRomualdi** [comment]
I cleaned the history and I also removed hardcoded 5 for the walking linear velocity.

The PR is ready to be reviewed @S-Dafarra and @kouroshD

**Review by S-Dafarra — APPROVED** [review]
LGTM. Just to understand, the connection of the goal port to the walking controller needs to be done manually, right?

**Comment by GiulioRomualdi** [comment]
Yes as before. In theory, this should be done by the application.

File: app/robots/iCubGenova09/virtualizerConfig.ini [hunk 1]
```diff
@@ -2,23 +2,23 @@ name            virtualizer
...
 #Minimum walking velocity to send a reference to the robot (value in meters per second)
-speed_deadzone        0.1
+speed_deadzone        0.5
 
 # RPC options
 playerOrientationPort_name    /playerOrientation:o
 robotOrientationPort_name     /robotOrientation:i
-rpcWalkingPort_name           /walkingRpc
-scale_X                       3.0
+rpcWalkingPort_name           /goal:o
```
Comment (kouroshD) [inline app/robots/iCubGenova09/virtualizerConfig.ini:13]: Please update the `rpcWalkingPort_name` with some name like `robotGoalPort_name` or sth similar.

Comment (kouroshD) [reply app/robots/iCubGenova09/virtualizerConfig.ini:13]: Please update the port name in all the xml files.

Comment (kouroshD) [reply app/robots/iCubGenova09/virtualizerConfig.ini:13]: Also, please also put the same port name for all the robot in the `.ini` files.
Thanks :)

**Review by kouroshD — CHANGES_REQUESTED** [review]
Thank you @GiulioRomualdi  for the PR. I have added some minor comments.

File: app/robots/iCubGenova09/virtualizerConfig.ini (hunk 1 above)
Comment (kouroshD) [inline app/robots/iCubGenova09/virtualizerConfig.ini:13]:
```suggestion
goalWalkingPort_name           /goal:o
```

File: modules/Virtualizer_module/src/VirtualizerModule.cpp [hunk 2]
```diff
@@ -261,12 +261,18 @@ bool VirtualizerModule::configure(yarp::os::ResourceFinder& rf)
         return false;
     }
 
+    if (!m_robotOrientationPort.open("/" + getName() + portName))
+    {
+        yError() << "[configure] " << portName << " port already open.";
+        return false;
+    }
+
     if (!YarpHelper::getStringFromSearchable(rf, "rpcWalkingPort_name", portName))
```
Comment (kouroshD) [inline modules/Virtualizer_module/src/VirtualizerModule.cpp:270]:
```suggestion
    if (!YarpHelper::getStringFromSearchable(rf, "goalWalkingPort_name", portName))
```

**Review by kouroshD — CHANGES_REQUESTED** [review]

**Comment by kouroshD** [comment]
@GiulioRomualdi  let me know if there are some updates on this?

**Comment by kouroshD** [comment]
@GiulioRomualdi  As I wrote you before, I have updated the configuration files; I rebased it also on top of devel :) .
Let me know if it is OK for you, so we can merge it.

**Review by kouroshD — APPROVED** [review]

**Comment by GiulioRomualdi** [comment]
> @GiulioRomualdi As I wrote you before, I have updated the configuration files; I rebased it also on top of devel :) .
> Please have a look as well, and let me know if it is OK for you, so we can merge it.

It's ok! You can merge it. Thank you @kouroshD

**Pull request merged**